components/properties/src/provider/bidi_data.rs

```diff
@@ -0,0 +1,265 @@
+                ty: "MirroredPairedBracketDataULE",
+                len: bytes.len(),
+            });
+        }
+        // Validate the bytes
+        #[allow(clippy::indexing_slicing)] // Won't panic because the chunks are always 3 bytes long
+        for byte_triple in bytes.chunks_exact(3) {
+            // Bidi_Mirroring_Glyph validation
+            let byte0 = byte_triple[0];
+            let byte1 = byte_triple[1];
+            let byte2 = byte_triple[2];
+            let mut mirroring_glyph_code_point: u32 = (byte2 & 0x1F) as u32;
+            mirroring_glyph_code_point = (mirroring_glyph_code_point << 8) | (byte1 as u32);
+            mirroring_glyph_code_point = (mirroring_glyph_code_point << 8) | (byte0 as u32);
+            let _mirroring_glyph = char::from_u32(mirroring_glyph_code_point)
+                .ok_or(ZeroVecError::VarZeroVecFormatError)?;
+
+            // skip validating the Bidi_Mirrored boolean since it is always valid
+
+            // assert that Bidi_Paired_Bracket_Type cannot have a 4th value because it only
+            // has 3 values: Open, Close, None
+            let paired_bracket_type_discriminant = byte2 >> 6;
+            if paired_bracket_type_discriminant == 0x3 {
+                return Err(ZeroVecError::ParseError {
+                    ty:
+                        "Unrecognized value for paired_bracket_type in MirroredPairedBracketDataULE",
+                });
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl AsULE for MirroredPairedBracketData {
+    type ULE = MirroredPairedBracketDataULE;
+
+    #[inline]
+    fn to_unaligned(self) -> Self::ULE {
+        let mirr_glyph_char_ule_ref_slice = &[self.mirroring_glyph.to_unaligned()];
+        let byte_slice = CharULE::as_byte_slice(mirr_glyph_char_ule_ref_slice);
+        let mut byte2 = byte_slice.get(2).copied().unwrap_or_default();
+        byte2 |= (self.mirrored as u8) << 5;
+        byte2 |= self.paired_bracket_type.0 << 6;
+
+        MirroredPairedBracketDataULE([
+            byte_slice.first().copied().unwrap_or_default(),
+            byte_slice.get(1).copied().unwrap_or_default(),
+            byte2,
+        ])
+    }
+
+    #[inline]
+    fn from_unaligned(unaligned: Self::ULE) -> Self {
+        // Safe because the lower bits 20..0 of MirroredPairedBracketDataULE bytes are the CharULE bytes,
+        // and CharULE::from_unaligned is safe because bytes are defined to represent a valid Unicode code point.
+        let mirroring_glyph = unsafe {
+            let mirroring_glyph_ule_bytes =
+                &[unaligned.0[0], unaligned.0[1], unaligned.0[2] & 0x1F];
+            let mirroring_glyph_ule = CharULE::from_byte_slice_unchecked(mirroring_glyph_ule_bytes);
+            mirroring_glyph_ule
+                .first()
+                .map(|ule| char::from_unaligned(*ule))
+                .unwrap_or(char::REPLACEMENT_CHARACTER)
+        };
+        let mirrored = (unaligned.0[2] >> 5) & 0x1 == 1;
```

Comment:
Nit (optional): I don't like having to look up operator precedence

```suggestion
        let mirrored = ((unaligned.0[2] >> 5) & 0x1) == 1;
```